src/array.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sparrow/array.hpp"
+#include "sparrow/array_factory.hpp"
+#include "sparrow/layout/array_helper.hpp"
+
+namespace sparrow
+{
+    array::array(ArrowArray&& array, ArrowSchema&& schema)
+        : p_array(array_factory(arrow_proxy(std::move(array), std::move(schema))))
```

Comment:
```suggestion
        : p_array(array_factory(arrow_proxy(std::forward<ArrowArray>(array), std::forward<ArrowSchemay>(schema))))
```

Reply:
`std::move` is correct here.
`std::forward` is for generic code, but here are not templates